kinesis-client/src/main/java/com/accenture/rig/RunnableKinesis.java

```diff
@@ -0,0 +1,80 @@
+package com.accenture.rig;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Optional;
+import java.util.UUID;
+
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
+import com.amazonaws.services.kinesis.clientlibrary.interfaces.v2.IRecordProcessorFactory;
+import com.amazonaws.services.kinesis.clientlibrary.lib.worker.InitialPositionInStream;
+import com.amazonaws.services.kinesis.clientlibrary.lib.worker.KinesisClientLibConfiguration;
+import com.amazonaws.services.kinesis.clientlibrary.lib.worker.Worker;
+import com.ericsson.otp.erlang.OtpAuthException;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+public final class RunnableKinesis implements Runnable {
+  private static final Log LOG = LogFactory.getLog(RunnableKinesis.class);
+
+  @Override
+  public void run() {
+    try {
+      doRun();
+    } catch (Throwable e) {
+      e.printStackTrace();
+    } finally {
+      // the worker has stopped, so let's bring down the VM:
+      System.exit(1);
+    }
+  }
+
+  public void doRun() {
+    final String rigErlangName = fetchEnv("RIG_ERLANG_NAME");
+    final String rigErlangCookie = fetchEnv("RIG_ERLANG_COOKIE");
+
+    LOG.info(String.format("Connecting to RIG (%s)..", rigErlangName));
+    ErlangInterface erlInterface = null;
+    try {
+      erlInterface = new ErlangInterface(rigErlangName, rigErlangCookie);
+    } catch (OtpAuthException | IOException e) {
+      e.printStackTrace();
+      System.exit(1);
+    }
+
+    // Ensure the JVM will refresh the cached IP values of AWS resources (e.g. service endpoints).
+    java.security.Security.setProperty("networkaddress.cache.ttl", "60");
+
+    String workerId = null;
+    try {
+      workerId = InetAddress.getLocalHost().getCanonicalHostName() + ":" + UUID.randomUUID();
+    } catch (UnknownHostException e) {
+      e.printStackTrace();
+      System.exit(1);
+    }
+    final String appName = fetchEnv("KINESIS_APP_NAME");
+    final String awsRegion = fetchEnv("KINESIS_AWS_REGION");
+    final String kinesisStream = fetchEnv("KINESIS_STREAM");
+    LOG.info(String.format("Kinesis config: app-name=%s aws-region=%s stream-name=%s", appName, awsRegion, kinesisStream));
+
+    final KinesisClientLibConfiguration config = new KinesisClientLibConfiguration(appName, kinesisStream,
+        new DefaultAWSCredentialsProviderChain(), workerId);
+    config.withRegionName(awsRegion);
+    config.withInitialPositionInStream(InitialPositionInStream.LATEST);
```

Comment:
in ARES you always send all messages down to the client, even the old (and possibly outdated) ones?

Reply:
Currently we do, but even if not I think it might be useful to be able to control such thing, e.g. if app goes down you may be interested in messages that happened some time ago.

Reply:
That's not the use case that RIG is build for, don't you think? I mean we never deal with lost messages and so on; would be strange to give the option to replay messages without the guarantee that the replayed messages actually end up anywhere...

Reply:
Hmm, not sure, I would say it's up to us, what we will enable for end users, but you can still achieve such message replay with Kafka if you set offsets or?

Reply:
sure, you can replay... what I mean is that due to network partitions and all kinds of things you might lose messages, also during replay, so you'll always need a way to re-request messages. If you have that, re-requesting is the better than replaying from horizon anyway, because stuff on the message queue might be outdated whereas a "re-request microservice" can send the newest data. To me that was part of the point of the architecture, but there might be different views on this